include/fplus/string_tools.hpp

```diff
@@ -125,6 +125,28 @@ String trim_whitespace(const String& str)
     return trim_by(is_whitespace<String>, str);
 }
 
+// API search type: to_lower_case : String -> String
+// to_lower_case("ChaRacTer&WorDs23") == "character&words23"
+template <typename String>
+std::string to_lower_case( const std::string& str ) {
+    return fplus::transform( [](char c) -> char{
+        if ( c >= 'A' && c <= 'Z' ) {
+            return c -'A' + 'a';
+        }
+    }, s );
+}
+
+// API search type: to_upper_case : String -> String
+// to_upper_case("ChaRacTer&WorDs34") == "CHARACTER&WORDS34"
+template <typename String>
+std::string to_upper_case( const std::string& str ) {
+    return fplus::transform( [](char c) -> char{
```

Comment:
Why not use `std::tolower`?  http://en.cppreference.com/w/cpp/string/byte/tolower